server/models/Types.ts

```diff
@@ -0,0 +1,18 @@
+import { ObjectID } from "mongodb";
+
+export interface Types {
+  _id: ObjectID;
+  name: string;
+  description?: string;
+  date?: Date;
+  start_time?: Date;
+  end_time?: Date;
+  location?: string;
+  start_registration?: boolean;
+  end_registration?: boolean;
```

Comment:
These should be start and end registration dates so they need the same type as `start_time` and `end_time`. The purpose of these fields is that we'll only allow a user to register if the current time falls in the `start_registration` and `end_registration` window.